blog/deployment.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: blog
+  labels:
+    app: blog
+spec:
+  serviceName: blog
+  replicas: 1
+  selector:
+    matchLabels:
+      app: blog
+  template:
+    metadata:
+      labels:
+        app: blog
+    spec:
+{% if deploy %}
+      priorityClassName: production
+{% endif %}
+      containers:
+      - name: blog
```

Comment:
You should include liveness and readiness probes.  You'll need to change the port and URL for ghost:

```
          livenessProbe:
            httpGet:
              path: /healthcheck
              port: 5000
            initialDelaySeconds: 5
            periodSeconds: 5
          readinessProbe:
            httpGet:
              path: /healthcheck
              port: 5000
            initialDelaySeconds: 5
            periodSeconds: 5
```

These are important so if the service gets stuck, k8s will restart, and when upgrading, k8s won't route traffic to the service until it can respond.  (Although it doesn't really matter for a single-instance stateful set, it is still best practice.)

Reply:
You should also include resources.  `requests` is the minimum resources that should be reserved for it, for example:

```
          resources:
            requests:
              memory: "1G"
              cpu: "300m"
```

`limits`, optional, is the maximum amount of resources the service can take advantage of.  We probably won't get a ton of traffic, so a fraction of a CPU like 300m is probably fine.  We'll want to revisit if we e.g. post something on hacker news.  I normally set memory by running the service for a while and watching how much memory it uses.  You can get the memory usage with `kubectl top pods`.